snuba/utils/streams/synchronized.py

```diff
@@ -43,7 +71,7 @@ class SynchronizedConsumer(Consumer[TPayload]):
     def __init__(
         self,
         consumer: Consumer[TPayload],
-        commit_log_consumer: Consumer[Commit],
+        commit_log_consumer: Consumer[KafkaPayload],
```

Comment:
Ideally, this would go back to using `Consumer[Commit]` with the codec wrapper approach in the description to decouple this from the `KafkaPayload`.

How I see this working is there would be a _new_ type of consumer with the signature `Consumer[TDecoded]`, which accepts as constructor parameters:

1. `Consumer[TEncoded]` 
2. `Codec[TEncoded, TDecoded]`

At this point, `Consumer.poll` would have a return type of `Optional[Message[TDecoded]]`.

For example:

```python
commit_consumer: Consumer[Commit] = CodecConsumer(
  KafkaConsumer(…),  # type: Consumer[KafkaPayload]
  commit_codec,  # type: Codec[KafkaPayload, Commit]
)
```